client/finality-grandpa/src/communication/gossip.rs

```diff
@@ -539,21 +542,26 @@ impl<N: Ord> Peers<N> {
 	}
 
 	fn authorities(&self) -> usize {
-		self.inner.iter().filter(|(_, info)| info.roles.is_authority()).count()
+		self.inner.iter().filter(|(_, info)| matches!(info.roles, ObservedRole::Authority)).count()
 	}
 
 	fn non_authorities(&self) -> usize {
-		self.inner.iter().filter(|(_, info)| !info.roles.is_authority()).count()
+		self.inner
+			.iter()
+			.filter(|(_, info)| matches!(info.roles, ObservedRole::Full | ObservedRole::Light))
```

Comment:
```suggestion
			.filter(|(_, info)| matches!(info.roles, ObservedRole::Full | ObservedRole::Light | ObservedRole::OurSentry))
```

Shouldn't our sentry be included here?

Reply:
It's purposefully not included.
This code decides who to send messages to. `OurSentry` and `OurGuardedAuthority` are excluded from this mechanism.

Reply:
Right. Sorry for the noise. Would you mind adding a comment for future me and @tomusdrw?

Reply:
I'd wait for someone like @andresilva to confirm that what I'm doing is indeed correct.

Reply:
That is correct :+1: